File: C#/Algorithms/String/LongestPalindrome.cs
/**
 * Class LongestPalindrome
 */
public class LongestPalindrome {

    /**
     * Finds the longest palindrome in a given string
     *
     * @param string str
     * @return string
     */
    public string findLongestPalindrome(string str) {
        IsPalindrome isPalindrome = new IsPalindrome();

        if (isPalindrome.isPalindromeIteration(str)) {
            return str;
        }

        int stringLength = str.Length;
        string longestPalindrome = "";
        int longestPalindromeLength = 1;

        for (int start = 0;start < stringLength - 1;start++) {
            for (int subStringLength = start + 2;subStringLength < stringLength - start + 1;subStringLength++) {
                string subString = str.Substring(start, subStringLength);
                if (isPalindrome.isPalindromeIteration(subString)) {
                    int tempSubStringLength = subString.Length;
                    if (tempSubStringLength > longestPalindromeLength) {
                        longestPalindrome = subString;
                        longestPalindromeLength = tempSubStringLength;
                    }
                }
            }
        }

        if (longestPalindromeLength == 1) {
            return str[0].ToString();
        }

        return longestPalindrome;
    }
}

File: PHP/Algorithms/String/longestPalindrome.php
<?php

require_once 'isPalindrome.php';

/**
 * Finds the longest palindrome in a given string
 *
 * @param string $string
 * @return string
 */
function findLongestPalindrome($string)
{
    if (isPalindromeIteration($string)) {
        return $string;
    }

    $stringLength = strlen($string);
    $longestPalindrome = '';
    $longestPalindromeLength = 1;

    for ($start = 0;$start < $stringLength - 1;$start++) {
        for ($subStringLength = 2;$subStringLength < $stringLength - $start + 1;$subStringLength++) {
            $subString = substr($string, $start, $subStringLength);
            if (isPalindromeIteration($subString)) {
                $tempSubStringLength = strlen($subString);
                if ($tempSubStringLength > $longestPalindromeLength) {
                    $longestPalindrome = $subString;
                    $longestPalindromeLength = $tempSubStringLength;
                }
            }
        }
    }

    if ($longestPalindromeLength == 1) {
        return $string[0];
    }

    return $longestPalindrome;
}
